client/app/lib/kodingappscontroller.coffee

```diff
@@ -29,7 +53,9 @@ module.exports = class KodingAppsController extends KDController
     app = globals.config.apps[name]
 
     @putAppScript app, (err, res) =>
-      AppClass = require res.app.identifier
+
+      AppClass = AppClasses[res.app.identifier] or kd.Object
```

Comment:
[AppController](https://github.com/koding/koding/blob/master/client/app/lib/appcontroller.coffee) would be a better default instead of `kd.Object`.